src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListFourth.java

```diff
@@ -0,0 +1,37 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+import edu.kis.powp.jobs2d.drivers.command.ComplexCommand;
+import edu.kis.powp.jobs2d.drivers.command.DriverCommand;
+import edu.kis.powp.jobs2d.drivers.command.OperateToCommand;
+import edu.kis.powp.jobs2d.drivers.command.SetPositionCommand;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SelectTestFigureOptionListFourth implements ActionListener {
+
+    private DriverManager driverManager;
+
+    public SelectTestFigureOptionListFourth(DriverManager driverManagerVariable) {
+        driverManager = driverManagerVariable;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        Job2dDriver driver = driverManager.getCurrentDriver();
+        List<DriverCommand> clist = new ArrayList<>();
+        clist.add(0, new SetPositionCommand(0, 0, driver));
+        clist.add(1, new OperateToCommand(100, 0, driver));
+        clist.add(2, new OperateToCommand(100, -50, driver));
+        clist.add(3, new OperateToCommand(0, -50, driver));
+        clist.add(4, new SetPositionCommand(100, -50, driver));
+        clist.add(5, new OperateToCommand(100, -100, driver));
+        clist.add(6, new OperateToCommand(0, -100, driver));
+
+        new ComplexCommand(clist).execute();
```

Comment:
This is not a factory. A factory should return a driver command (not execute it). Missing point 3.1.5.